Automattic/woocommerce-payments
# Dev: Move Klarna E2E tests to Playwright

Fixes #9376

#### Changes proposed in this Pull Request

The Klarna E2E tests are currently difficult to maintain given that the contents of the iframes or pages that we are currently using in the flows, are completely outside of our control.

A recent example of an external change breaking the tests can be found here: https://github.com/Automattic/woocommerce-payments/issues/9339#top

This PR does the following:
* Migrate the Klarna tests from Puppeteer to Playwright.
* Reduce the actions performed in Klarna Playground, given the unpredictability of the rendered elements that are completely outside of our control.
* Improve the stability of clicking the place order button in Playwright, which is a common source of flaky tests.

#### Testing instructions

* Follow the instructions in [/tests/e2e-pw/README.md](https://github.com/Automattic/woocommerce-payments/blob/81e3207c2ee13c3d5dc68b40839e984992c5ea12/tests/e2e-pw/README.md) to run the tests locally.
* Alternatively, you can verify the Playwright results in this PR checks and also run the Puppeteer E2E tests in a github action for this branch in the following URL https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml

-------------------


**Post merge**
